### PR TITLE
Add Azure Log Analytics examples

### PR DESCRIPTION
**Pull request by ading1977**
This PR adds reference implementations (in GoLang and Python) of a metric server to pull in memory metrics from Azure Log Analytics workspace via Azure Log Analytics REST API.

### CODE REVIEW

File: example/azure-loganalytics/deploy/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            - name: AZURE_LOG_ANALYTICS_WORKSPACES
+              value: <WORKSPACE_IDS_SEPARATED_BY_COMMA>
+            - name: AZURE_TENANT_ID
+              value: <TENANT_ID>
+            - name: AZURE_CLIENT_ID
+              valueFrom:
+                secretKeyRef:
+                  name: azure-loganalytics-clientid
+                  key: clientid
+            - name: AZURE_CLIENT_SECRET
```
Comment (esara) [inline example/azure-loganalytics/deploy/deploy.yaml:31]: do you think we can share the azure client secret with
https://github.com/turbonomic/t8c-install/blob/master/operator/helm-charts/probes/mediation-azure/templates/mediation-azure.yaml#L107

Comment (ading1977) [reply example/azure-loganalytics/deploy/deploy.yaml]: After discussing with @chlam4, I have modified the way we obtain azure account information to be the same with all azure probes through the `azure` secret. For `azure-loganalytics` metric server,  for simplicity, we require the user to specify a specific target ID, i.e., we only authenticate with one account.

Comment (ading1977) [reply example/azure-loganalytics/deploy/deploy.yaml]: Updated the deployment instructions in the [README](https://github.com/turbonomic/data-ingestion-framework/blob/c402068d48df0954d540d167680f6ad6331dca08/example/azure-loganalytics/README.md).

**Comment by ading1977** [comment]
I have updated the PR [21381aa](https://github.com/turbonomic/data-ingestion-framework/pull/24/commits/21381aaebdfe4c0c9b35e61800e1b517f07274c2) to use the `turbonomic/azure-loganalytics-python` image.

**Pull request merged**